### PR TITLE
Migrate fxapom to pytest-selenium

### PR DESCRIPTION
**Pull request by bobsilverberg**
It looks like I may need to make some changes to FxAPom to support a new test for Markteplace, so I decided to do some cleanup first. Step 1 is to migrate to pytest-selenium, after which I will also remove all implicit waits as well as clean up the page objects a bit more.

I had to add one explicit wait after switching to pytest-selenium to allow the tests to pass.

There is an adhoc scheduled at http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/job/fxapom.adhoc.pytest-selenium/, but it's currently at the end of a long queue.

@mozilla/web-qa-sorcerers @davehunt r?

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -8,14 +8,19 @@
 from fxapom.fxapom import DEV_URL, FxATestAccount, PROD_URL
 
 
-def pytest_funcarg__mozwebqa(request):
+@pytest.fixture(scope='session')
+def capabilities(capabilities):
+    capabilities.setdefault('tags', []).append('oneanddone')
```
Comment (davehunt) [inline conftest.py]: This should be 'fxapom'.

File: conftest.py [hunk 2]
```diff
@@ -8,14 +8,19 @@
...
+
+
+@pytest.fixture
+def click_login(base_url, selenium):
     fxa_login_button_locator_css = 'button.signin'
-    mozwebqa = request.getfuncargvalue('mozwebqa')
-    mozwebqa.selenium.get('%s/' % mozwebqa.base_url)
-    WebDriverWait(mozwebqa.selenium, mozwebqa.timeout).until(
+    selenium.get('%s/' % base_url)
+    WebDriverWait(selenium, 10).until(
```
Comment (davehunt) [inline conftest.py]: A later refactor could remove this hard-coded timeout, but let's leave it for now.

File: fxapom/pages/page.py [hunk 3]
```diff
@@ -34,7 +33,7 @@ def is_element_present(self, *locator):
             return False
         finally:
             # set the implicit wait back
-            self.selenium.implicitly_wait(self.testsetup.default_implicit_wait)
+            self.selenium.implicitly_wait(self.timeout)
```
Comment (davehunt) [inline fxapom/pages/page.py]: We should avoid setting an implicit wait, unless we set an initial implicit wait.

File: tests/test_login.py [hunk 4]
```diff
@@ -12,8 +12,8 @@ class TestLogin(object):
 
     _fxa_logged_in_indicator_locator = (By.ID, 'loggedin')
 
-    def test_user_can_sign_in(self, mozwebqa, dev_account):
-        fxa = WebDriverFxA(mozwebqa)
+    def test_user_can_sign_in(self, base_url, selenium, dev_account, click_login):
+        fxa = WebDriverFxA(base_url, selenium)
         fxa.sign_in(dev_account.email, dev_account.password)
-        WebDriverWait(mozwebqa.selenium, mozwebqa.timeout).until(
+        WebDriverWait(selenium, 20).until(
```
Comment (davehunt) [inline tests/test_login.py]: Why the higher timeout here? As we have several hard-coded timeouts, I'm wondering if we should add a `timeout` fixture so we can set this in just one location.

**Comment by davehunt** [comment]
There were issues with the adhoc job configuration. I fixed them and reran: http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/job/fxapom.adhoc.pytest-selenium/3/

File: conftest.py (hunk 1 above)
Comment (bobsilverberg) [inline conftest.py]: Thanks, fixed!

File: fxapom/pages/page.py (hunk 3 above)
Comment (bobsilverberg) [inline fxapom/pages/page.py]: Yes, I know. I was trying to be "good" and not include too many changes in a single PR. The idea was to only do what was absolutely necessary to get this running with pytest-selenium and then do another PR afterward which addresses implicit waits and some other page object changes. Do you think it's okay to leave this as is for this PR?

File: tests/test_login.py (hunk 4 above)
Comment (bobsilverberg) [inline tests/test_login.py]: When I was testing locally I saw a failure once because the login was taking so long to complete that I got a timeout here with `10`. I don't think we need to change all timeouts to `20`, just this one, so I wouldn't want to add a fixture with a value of `20`.

I'm not quite as concerned with hardcoded timeouts in this project, tbh, as there are only a handful of tests and they really only exist to prove that fxapom works. I think it's fine to special case tests when needed because of that.

**Comment by bobsilverberg** [comment]
Thanks for the review @davehunt. I fixed the tags capability issue and have kicked off another adhoc (which is currently queued). As you will see from my comments above, I don't plan on making any other changes in this PR to address your comments, but I can if you feel strongly that something should be changed. Please let me know.

File: fxapom/pages/page.py (hunk 3 above)
Comment (davehunt) [inline fxapom/pages/page.py]: Sure, makes sense.

File: tests/test_login.py (hunk 4 above)
Comment (davehunt) [inline tests/test_login.py]: Fair enough, let's address this in the future if it ever becomes an issue.

**Comment by davehunt** [comment]
r+
